x-pack/auditbeat/processors/sessionmd/provider/procfs_provider/procfs_provider_test.go

```diff
@@ -0,0 +1,452 @@
+						Pgid:        0,
+						Sid:         1,
+					},
+				},
+				{
+					PIDs: types.PIDInfo{
+						StartTimeNS: 0,
+						Tid:         200,
+						Tgid:        200,
+						Vpid:        0,
+						Ppid:        100,
+						Pgid:        0,
+						Sid:         100,
+					},
+				},
+			},
+			procinfo: []procfs.ProcessInfo{},
+			expected: procfs.ProcessInfo{
+				PIDs: types.PIDInfo{
+					Tid:  200,
+					Tgid: 200,
+					Ppid: 100,
+					Pgid: 0,
+					Sid:  200,
+				},
+			},
+		},
+		{
+			// Test case: Setsid failed
+			pid: 200,
+			event: beat.Event{
+				Timestamp: timestamp,
+				Fields: mapstr.M{
+					"auditd": mapstr.M{
+						"data": mapstr.M{
+							"a0":      "1",
+							"a1":      "ffffefbfcb78",
+							"a2":      "ffffefbfcb88",
+							"a3":      "410134",
+							"arch":    "aarch64",
+							"exit":    "EPERM",
+							"syscall": "setsid",
+							"tty":     "pts4",
+						},
+						"result": "fail",
+					},
+					"process": mapstr.M{
+						"pid": 200,
+						"parent": mapstr.M{
+							"pid": 100,
+						},
+					},
+				},
+			},
+			prereq: []procfs.ProcessInfo{
+				{
+					PIDs: types.PIDInfo{
+						StartTimeNS: 0,
+						Tid:         100,
+						Tgid:        100,
+						Vpid:        0,
+						Ppid:        0,
+						Pgid:        0,
+						Sid:         1,
+					},
+				},
+				{
+					PIDs: types.PIDInfo{
+						StartTimeNS: 0,
+						Tid:         200,
+						Tgid:        200,
+						Vpid:        0,
+						Ppid:        100,
+						Pgid:        0,
+						Sid:         100,
+					},
+				},
+			},
+			procinfo: []procfs.ProcessInfo{},
+			expected: procfs.ProcessInfo{
+				PIDs: types.PIDInfo{
+					Tid:  200,
+					Tgid: 200,
+					Ppid: 100,
+					Pgid: 0,
+					Sid:  100,
+				},
+			},
+		},
+		{
+			// Test case: Session leader not scraped
+			pid: 200,
+			event: beat.Event{
+				Timestamp: timestamp,
+				Fields: mapstr.M{
+					"auditd": mapstr.M{
+						"data": mapstr.M{
+							"a0":      "1",
+							"a1":      "ffffeb535e38",
+							"a2":      "ffffeb535e48",
+							"a3":      "410134",
+							"arch":    "aarch64",
+							"exit":    "200",
+							"syscall": "setsid",
+							"tty":     "pts4",
+						},
+						"result": "success",
+					},
+					"process": mapstr.M{
+						"pid": 200,
+						"parent": mapstr.M{
+							"pid": 100,
+						},
+					},
+				},
+			},
+			prereq: []procfs.ProcessInfo{
+				{
+					PIDs: types.PIDInfo{
+						StartTimeNS: 0,
+						Tid:         200,
+						Tgid:        200,
+						Vpid:        0,
+						Ppid:        100,
+						Pgid:        0,
+						Sid:         100,
+					},
+				},
+			},
+			procinfo: []procfs.ProcessInfo{},
+			expected: procfs.ProcessInfo{
+				PIDs: types.PIDInfo{
+					Tid:  200,
+					Tgid: 200,
+					Ppid: 100,
+					Pgid: 0,
+					Sid:  200,
+				},
+			},
+		},
+	}
+)
+
+func TestExecEvent(t *testing.T) {
+	for _, tt := range execTests {
+		tt := tt // avoids taking address of loop variable within loop
+		reader := procfs.NewMockReader()
```

Comment:
Let's do a separate `t.Run()` for each test case. You already have comments describing the test. Turn those into names and then each test case can be run in independently.

Reply:
I've separated the tests into a function for each